File: Documentacao/LIA2_008_PedagogicalAdaptive_Mobile_Thinking_FIX02/apps/backend/app/contracts/documentStructureContract.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel


class DocumentBlockViewContract(BaseModel):
    contractName: Literal["DocumentBlockView.v1"] = "DocumentBlockView.v1"
    documentBlockId: UUID
    sequenceNumber: int
    blockType: str
    textContent: str | None
    processingStatus: str
    orientationDegrees: int | None = None
    visionModelId: str | None = None
    visionThinkingEnabled: bool | None = None


class DocumentPageViewContract(BaseModel):
    contractName: Literal["DocumentPageView.v1"] = "DocumentPageView.v1"
    documentPageId: UUID
    pageNumber: int
    nativeText: str | None
    status: str
    blocks: list[DocumentBlockViewContract]


class DocumentStructureContract(BaseModel):
    contractName: Literal["DocumentStructure.v2"] = "DocumentStructure.v2"
    documentId: UUID
    documentVersionId: UUID
    extractionStatus: str
    pageCount: int
    pages: list[DocumentPageViewContract]
    evidenceCount: int
    chunkCount: int
    embeddedChunkCount: int
    visualPendingCount: int

File: Documentacao/LIA2_008_PedagogicalAdaptive_Mobile_Thinking_FIX02/apps/studentWeb/src/contracts/documentStructureContract.ts
export interface DocumentBlockViewContract {
  contractName: 'DocumentBlockView.v1'
  documentBlockId: string
  sequenceNumber: number
  blockType: string
  textContent: string | null
  processingStatus: string
  orientationDegrees: number | null
  visionModelId: string | null
  visionThinkingEnabled: boolean | null
}

export interface DocumentPageViewContract {
  contractName: 'DocumentPageView.v1'
  documentPageId: string
  pageNumber: number
  nativeText: string | null
  status: string
  blocks: DocumentBlockViewContract[]
}

export interface DocumentStructureContract {
  contractName: 'DocumentStructure.v2'
  documentId: string
  documentVersionId: string
  extractionStatus: string
  pageCount: number
  pages: DocumentPageViewContract[]
  evidenceCount: number
  chunkCount: number
  embeddedChunkCount: number
  visualPendingCount: number
}
